tests/spare/spare_job_test.py

```diff
@@ -0,0 +1,390 @@
+import boto3
+from botocore.response import StreamingBody
+from botocore.stub import ANY, Stubber
+import datetime
+import gzip
+import io
+import json
+import os
+import polars as pl
+from polars.testing import assert_frame_equal
+import pytest
+from unittest.mock import patch
+
+import odin.spare.spare_job as spare_job
+
+
+@pytest.fixture
+def s3_stub():
+    s3_client = boto3.client("s3")
+    with Stubber(s3_client) as stubber:
+        with patch("odin.utils.aws.s3.get_client", return_value=s3_client):
+            yield stubber
+        stubber.assert_no_pending_responses()
+
+
+api_endpoint: spare_job.ApiEndpoint = {
+    "name": "testname",
+    "api_path": "testpath",
+    "columns": {
+        "id": pl.String(),
+        "value": pl.Int32(),
+    },
+}
+
+
+def write_gzipped_json(data, path):
+    os.makedirs(os.path.dirname(path), exist_ok=True)
+    with open(path, "wb") as f:
+        f.write(gzip.compress(json.dumps(data).encode()))
+
+
+# load_parquet
+
+empty_df = pl.DataFrame(
+    {},
+    spare_job.METADATA_COLUMNS | api_endpoint["columns"],
+)
+one_row_df = pl.DataFrame(
+    {
+        "ROW_TIMESTAMP": [1745582400000],
+        "ROW_DELETED": [False],
+        "id": ["id"],
+        "value": [10],
+    },
+    spare_job.METADATA_COLUMNS | api_endpoint["columns"],
+)
+
+
+def test_load_parquet_s3_exists(s3_stub, tmpdir):
+    localpath = os.path.join(tmpdir, "test.parquet")
+    one_row_df.write_parquet(localpath)
+    with open(localpath, "rb") as f:
+        bytes = f.read()
+    s3_stub.add_response(
+        "head_object",
+        expected_params={
+            "Bucket": "bucket",
+            "Key": "test.parquet",
+        },
+        service_response={},
+    )
+    s3_stub.add_response(
+        "get_object",
+        expected_params={"Bucket": "bucket", "Key": "test.parquet"},
+        service_response={"Body": StreamingBody(io.BytesIO(bytes), len(bytes))},
+    )
+    df = spare_job.load_parquet("s3://bucket/test.parquet", api_endpoint["columns"])
+    assert_frame_equal(df, one_row_df)
+
+
+def test_load_parquet_s3_not_exists(s3_stub):
+    s3_stub.add_client_error(
+        "head_object",
+        expected_params={
+            "Bucket": "target",
+            "Key": "testname.parquet",
+        },
+        service_error_code="404",
+    )
+    df = spare_job.load_parquet("s3://target/testname.parquet", api_endpoint["columns"])
+    assert_frame_equal(df, empty_df)
+
+
+def test_load_parquet_local_exists(tmpdir):
+    path = os.path.join(tmpdir, "test.parquet")
+    one_row_df.write_parquet(path)
+    df = spare_job.load_parquet(path, api_endpoint["columns"])
+    assert_frame_equal(df, one_row_df)
+
+
+def test_load_parquet_local_not_exists(tmpdir):
+    path = os.path.join(tmpdir, "test.parquet")
+    df = spare_job.load_parquet(path, api_endpoint["columns"])
+    assert_frame_equal(df, empty_df)
+
+
+# datetime_from_delta_path_ms
+
+
+def test_datetime_from_delta_path_ms():
```

Comment:
CI is failing cuz some of my tests don't have a docstring. I don't think a docstring would add much that's not covered in the test name or easy to see from the implementation, (in this case, the docstring would be `"""It works."""`) so I'd rather skip them. Is it possible to turn off the required docstring check on test files?

Reply:
Added docstrings in [c575e2a](https://github.com/mbta/odin/pull/20/commits/c575e2a3924cddfba6e2906c525447ddd42ccfec)